x/rollup/keeper/msg_server_test.go

```diff
@@ -0,0 +1,236 @@
+package keeper_test
+
+import (
+	"encoding/json"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/golang/mock/gomock"
+	"github.com/polymerdao/monomer/testutils"
+	"github.com/polymerdao/monomer/x/rollup/keeper"
+	"github.com/polymerdao/monomer/x/rollup/types"
+)
+
+func (s *KeeperTestSuite) TestApplyL1Txs() {
+	l1AttributesTx, depositTx, cosmosEthTx := testutils.GenerateEthTxs(s.T())
+	// The only constraint for a contract creation tx is that it must be a non-system DepositTx with no To field
+	contractCreationTx := gethtypes.NewTx(&gethtypes.DepositTx{})
+
+	l1AttributesTxBz, err := l1AttributesTx.MarshalBinary()
+	s.Require().NoError(err)
+	depositTxBz, err := depositTx.MarshalBinary()
+	s.Require().NoError(err)
+	cosmosEthTxBz, err := cosmosEthTx.MarshalBinary()
+	s.Require().NoError(err)
+	contractCreationTxBz, err := contractCreationTx.MarshalBinary()
+	s.Require().NoError(err)
+	invalidTxBz := []byte("invalid tx bytes")
+
+	tests := map[string]struct {
+		msg         *types.MsgApplyL1Txs
+		setupMocks  func()
+		shouldError bool
+	}{
+		"successful message with single user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz},
+			},
+			shouldError: false,
+		},
+		"successful message with multiple user deposit txs": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, depositTxBz, depositTxBz},
+			},
+			shouldError: false,
+		},
+		"invalid l1 attributes tx bytes": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{invalidTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"non-deposit tx passed in as l1 attributes tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{cosmosEthTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"user deposit tx passed in as l1 attributes tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{depositTxBz, depositTxBz},
+			},
+			shouldError: true,
+		},
+		"invalid user deposit tx bytes": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, invalidTxBz},
+			},
+			shouldError: true,
+		},
+		"non-deposit tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, cosmosEthTxBz},
+			},
+			shouldError: true,
+		},
+		"l1 attributes tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, l1AttributesTxBz},
+			},
+			shouldError: true,
+		},
+		"contract creation tx passed in as user deposit tx": {
+			msg: &types.MsgApplyL1Txs{
+				TxBytes: [][]byte{l1AttributesTxBz, contractCreationTxBz},
+			},
+			shouldError: true,
+		},
+		"one successful l1 user deposit tx and an invalid tx passed in as user deposit txs": {
```

Comment:
nit: not sure if this is correct, but seems right
```suggestion
		"one valid l1 user deposit tx and an invalid tx passed in as user deposit txs": {
```